azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/JoinCallButtonHolderView.kt

```diff
@@ -63,11 +84,79 @@ internal class JoinCallButtonHolderView : ConstraintLayout {
         }
     }
 
+    fun stop() {
+        if (snackBar.isShown) {
+            snackBar.dismiss()
+        }
+        rootView.invalidate()
+        // to fix memory leak
+        snackBar.anchorView = null
+    }
+
     private fun onJoinCallEnabledChanged(isEnabled: Boolean) {
         setupJoinCallButton.isEnabled = isEnabled
         setupJoinCallButtonText.isEnabled = isEnabled
     }
 
+    private fun displaySnackBar() {
+        val errorMessage = context.getString(R.string.azure_communication_ui_calling_no_connection_available)
+        if (errorMessage.isBlank()) return
+        snackBarTextView.text = errorMessage
+        snackBar.run {
+            if (isShown) {
+                dismiss()
+            }
+            show()
+
+            view.contentDescription =
+                "${context.getString(R.string.azure_communication_ui_calling_alert_title)}: $errorMessage"
+            view.accessibilityFocus()
+        }
+    }
+
+    private fun initSnackBar() {
+        snackBar = Snackbar.make(
+            rootView,
+            "",
+            Snackbar.LENGTH_INDEFINITE,
+            Snackbar.Style.REGULAR
+        ).apply {
+            animationMode = BaseTransientBottomBar.ANIMATION_MODE_FADE
+            setAction(rootView.context!!.getText(R.string.azure_communication_ui_calling_snack_bar_button_dismiss)) {}
+            anchorView =
+                rootView.findViewById(R.id.azure_communication_ui_setup_join_call_button)
+            view.background.colorFilter = PorterDuffColorFilter(
+                ContextCompat.getColor(
+                    rootView.context,
+                    R.color.azure_communication_ui_calling_color_snack_bar_background
+                ),
+                PorterDuff.Mode.SRC_IN
+            )
+            snackBarTextView = view.findViewById(R.id.snackbar_text)
+            snackBarTextView.setTextColor(
+                ContextCompat.getColor(
+                    rootView.context,
+                    R.color.azure_communication_ui_calling_color_snack_bar_text_color
+                )
+            )
+            view.findViewById<AppCompatButton>(R.id.snackbar_action).apply {
+                setTextColor(
+                    ContextCompat.getColor(
+                        rootView.context,
+                        R.color.azure_communication_ui_calling_color_snack_bar_text_color
+                    )
+                )
+                isAllCaps = false
+                contentDescription =
+                    rootView.context.getText(R.string.azure_communication_ui_calling_snack_bar_button_dismiss)
+            }
+            ViewCompat.setImportantForAccessibility(
+                view,
+                ViewCompat.IMPORTANT_FOR_ACCESSIBILITY_YES
+            )
+        }
+    }
```

Comment:
We might have snackbar initialization code in the project elsewhere as the ui design is similar. If that is not reusable, how about making a snackbar component and simply reuse it everywhere in the composite?